tests/test_unmocked.py

```diff
@@ -0,0 +1,116 @@
+"""
+Handles unmocked testing.
+"""
+
+import os
+import unittest
+from dotenv import load_dotenv
+import fdc  # pylint: disable=import-error
+
+load_dotenv()
+fdc.set_key(os.getenv("FDC_KEY"))
+
+
+class TestFdcAPI(unittest.TestCase):
+    """Contains test cases for the FoodData Central API."""
+
+    def setUp(self):
+        self.fdc_id1 = 2057648  # cheddar cheese
+        self.fdc_id2 = 534358  # nut n' berry mix
+
+    def test_invalid_food(self):
+        """Trying to retrieve an invalid food returns an error"""
+
+        response = fdc.get_food_by_id(-1)
+
+        self.assertIn("success", response)
+        self.assertIn("error", response)
+        self.assertIn("code", response["error"])
+        self.assertIn("message", response["error"])
+
+        self.assertEqual(response["success"], False)
+        self.assertEqual(response["error"]["code"], "REQUEST_EXCEPTION")
+
+    def test_retrieve_food(self):
+        """Retrieving a food by FDC ID returns proper results"""
+
+        response = fdc.get_food_by_id(self.fdc_id1)
+
+        self.assertIn("success", response)
+        self.assertIn("fdcId", response)
+        self.assertIn("description", response)
+        self.assertIn("ingredients", response)
+
+        self.assertEqual(response["success"], True)
+        self.assertEqual(response["fdcId"], self.fdc_id1)
+        self.assertEqual(response["description"], "CHEDDAR CHEESE")
+
+    def test_retrieve_food_with_nutrient_ids(self):
+        """Retrieving a food with specific nutrient IDs returns only nutrients with those IDS."""
+
+        response = fdc.get_food_by_id(self.fdc_id1, nutrient_numbers=[301])
+
+        self.assertIn("success", response)
+        self.assertEqual(response["success"], True)
+
+        self.assertIn("foodNutrients", response)
+        self.assertEqual(len(response["foodNutrients"]), 1)
+
+        self.assertIn("nutrient", response["foodNutrients"][0])
+        self.assertIn("number", response["foodNutrients"][0]["nutrient"])
+
+        self.assertEqual(response["foodNutrients"][0]["nutrient"]["number"], "301")
+
```

Comment:
Not sure how the API works in returning data but is there a way to create a function to test retrieving a nutrient that does not exist? And is there a way to also try retrieving muliple nutrients to return proper results?

Reply:
Good idea, I'll edit `test_retrieve_food_with_nutrient_ids` to test for at least two nutrients and add a test for an undefined nutrient. I think the result of the latter should be a response with an empty `"foodNutrients"` array. Thanks!